Userland/Libraries/LibWeb/Painting/ViewportPaintable.cpp

```diff
@@ -303,10 +303,28 @@ void ViewportPaintable::recompute_selection_states()
     auto* start_container = range->start_container();
     auto* end_container = range->end_container();
 
-    // 3. If the selection starts and ends in the same node:
+    // 3. Mark the nodes included in range selected.
+    for (auto* node = start_container; node && node != end_container->next_in_pre_order(); node = node->next_in_pre_order()) {
+        if (auto* paintable = node->paintable())
+            paintable->set_selected(true);
+    }
+}
```

Comment:
> I am testing the branch locally and found that it breaks selection on pages where it used to work. you can reproduce the problem by trying to select a text using mouse on a page as simple as `<p>Hello</p>`.
> 
> Besides that issue the change seems good and thank you for making progress on it!

The problem is caused by incorrectly iterating over the range's included nodes, which is addressed here.